R/bootstrap.R

```diff
@@ -42,16 +42,60 @@ bootstrapPage <- function(..., title = NULL, responsive = NULL, theme = NULL) {
       # remainder of tags passed to the function
       list(...)
     ),
-    if (useBsTheme()) bootstraplib::bootstrap() else bootstrapLib()
+    bootstrapLib_()
   )
 }
 
+#' Bootstrap libraries
+#'
+#' This function is deprecated in favor of [bootstraplib::bootstrap()], which
+#' provides a more fully featured approach to accessing and theming Bootstrap
+#' CSS. Note that most users won't need to call this function directly since
+#' [bootstrapPage()] and it's special cases ( e.g., [fluidPage()],
+#' [basicPage()], [navbarPage()], etc) implicitly depend on Bootstrap.
+#'
+#' @param theme deprecated (it never worked).
+#' @export
+#' @keywords internal
+bootstrapLib <- function(theme = NULL) {
+  if (!is.null(theme)) {
+    stop("shiny::bootstrapLib()'s theme argument never worked as intended, don't use it.")
+  }
+  shinyDeprecated("bootstraplib::bootstrap()", old = "shiny::bootstrapLib()")
```

Comment:
I think using `shinyDeprecated` is too aggressive -- people who are using HTML templates will upgrade their packages or install them on a new system, and all of a sudden they'll start getting the deprecation message and be alarmed or confused by it, and they'll feel like it's kind of an emergency and they have to fix their apps right away. (I've had that happen with other packages and it's always a disconcerting experience.)  I think it's better to just allow it to work without a message, and update our docs to suggest using `bootstraplib::bootstrap()`.

We could leave this in, but comment it out, and add a comment with the current date and a reminder that we can un-comment the line in, say a year or two.

Reply:
This deprecation message will only happen if users are calling `bootstrapLib()` directly (internally we use `bootstrapLib_()`  which won't throw the message -- I don't think that's too aggressive?)

Reply:
In our docs for HTML templates, we suggest that they call `bootstrapLib()`, and there are probably a lot of apps out there that do this.
https://shiny.rstudio.com/articles/templates.html